Frontend and integration with backend for Task Add Screen

stirleng opened the pull request:
Barebones for Task Add screen based on Figma. Commit contains:

- DB query debugging function getEventByName in database.js
- JSX in CreateTaskScreen takes input and makes a call to addEvent in database.js
- helper function formatDate in CreateTaskScreen

Does NOT contain input validation or sanitization.

Tested:
- Adding events/tasks through app working

Untested: 
- Invalid data types as input

stirleng committed:
boilerplate text entry

stirleng committed:
CreateTaskScreen alpha

does not yet update sqlite database

stirleng committed:
Adding Task Screen Barebones Complete

Integrated with adding tasks/events on backend

emmatliu commented on GoNOW/app/screens/CreateTaskScreen.tsx:
```diff
@@ -1,8 +1,182 @@
+import DateTimePicker from '@react-native-community/datetimepicker';
+import { addEvent } from '../scripts/database';
 
+const TaskForm = () => {
```
this is kind of a nit but i would prefer that the default export in CreateTaskScreen be CreateTaskScreen.  did some cursory investigation of react-native naming standards and i think they suggest -form if it can be reused (and then it can be housed in a separate /components folder and used in the screen) but afaik we will only use this once.  open to hearing your thoughts on this ofc

stirleng replied:
yeah that's more intuitive, changing this

emmatliu commented on GoNOW/app/screens/CreateTaskScreen.tsx:
```diff
@@ -1,8 +1,182 @@
+      )}
+
+      <Text style={styles.label}>Enter transportation mode</Text>
+      <TextInput
```
I think this makes more sense as a dropdown.

stirleng replied:
Good point

emmatliu commented on GoNOW/app/screens/CreateTaskScreen.tsx:
```diff
@@ -1,8 +1,182 @@
+        <Text style={styles.label}>Autoschedule?</Text>
+      </View>
+
+      <Text style={styles.label}>Enter latitude</Text>
```
Not sure if I'm understanding your code correctly, but are users being asked to put in long/lat of location? Can we use this instead: https://nominatim.org/

stirleng replied:
Yes, as of right now. Maybe in a later commit? This is just for preliminary testing

emmatliu commented on GoNOW/app/screens/CreateTaskScreen.tsx:
```diff
@@ -1,8 +1,182 @@
+  );
+};
+
+const styles = StyleSheet.create({
```
another style nit -- would like this to be explicitly a .ts stylesheet that is imported, in my miri/daily-tasks you can see how i set it up as a /screen_styles under /screens (i am open to other thoughts on how to set this up as well)

WalterChen352 replied:
I am onboard with this practice of separating styles and the component.

stirleng replied:
Will do

emmatliu commented on GoNOW/app/index.tsx:
I def think what you commented should be removed in the master branch (perhaps in a quick fix pr).  I don't think it goes in this pr though so if you could remove this file from your changes that would be great

stirleng committed:
TransportationMode dropdown

untested

stirleng committed:
pr fixes

Discussion under "Frontend and integration with backend for Task Add Screen" pr

stirleng committed:
typescript changes

stirleng committed:
Merge branch 'master' into task-add-frontend

stirleng committed:
linter fixes

stirleng committed:
linter fixes

WalterChen352 reviewed (approved):
Looks good 👍
